src/treecodes.py

```diff
@@ -1,238 +1,58 @@
-             'other': {'group': 'T', 'code': 'o'},
-             'other-e10s': {'group': 'T-e10s', 'code': 'o'},
-             'other_nol64': {'group': 'T', 'code': 'o'},
-             'other-e10s_nol64': {'group': 'T-e10s', 'code': 'o'},
-             'other_l64': {'group': 'T', 'code': 'o'},
-             'other-e10s_l64': {'group': 'T-e10s', 'code': 'o'},
-             'svgr': {'group': 'T', 'code': 's'},
-             'svgr-e10s': {'group': 'T-e10s', 'code': 's'},
-             'svgr-osx-e10s': {'group': 'T-e10s', 'code': 's'},
-             'g1': {'group': 'T', 'code': 'g1'},
-             'g1-e10s': {'group': 'T-e10s', 'code': 'g1'},
-             'g1-osx-e10s': {'group': 'T-e10s', 'code': 'g1'},
-             'g2': {'group': 'T', 'code': 'g2'},
-             'g2-e10s': {'group': 'T-e10s', 'code': 'g2'},
-             'g2-osx-e10s': {'group': 'T-e10s', 'code': 'g2'},
-             'dromaeojs': {'group': 'T', 'code': 'd'},
-             'dromaeojs-e10s': {'group': 'T-e10s', 'code': 'd'},
-             'chromez': {'group': 'T', 'code': 'c'},
-             'chromez-e10s': {'group': 'T-e10s', 'code': 'c'},
-             'chromez-osx-e10s': {'group': 'T-e10s', 'code': 'c'},
-             'xperf': {'group': 'T', 'code': 'x'},
-             'xperf-e10s': {'group': 'T-e10s', 'code': 'x'},
-             'androidx86-set-4': {'group': 'S', 'code': '4'}}
+import requests
 
 
-def getGroup(name):
-    try:
-        group = name.split('-')[0]
-    except:
-        group = ''
-    return group
+class Treecodes:
+    def __init__(self, repo_name='mozilla-inbound'):
+        # default to query all jobs on mozilla-inbound branch
+        self.tbplnames = {}
+        url = 'https://treeherder.mozilla.org/api/project/%s/runnable_jobs/?format=json'
+        data = requests.get(url % repo_name).json()
+        joblist = data['results']
+        # skipping pgo - We run this infrequent enough that we should have all pgo results tested
+        self.joblist = [job for job in joblist if job['platform_option'] != 'pgo']
+        if len(self.joblist) > 0:
+            for job in self.joblist:
+                # the format of buildbot job is different from taskcluter jbo
+                if job['build_system_type'] == 'buildbot':
+                    testtype = job['ref_data_name'].split(' ')[-1]
+                    job_group_symbol = job['job_group_symbol'] \
+                        if job['job_group_symbol'] != '?' else ''
+                    job_type_symbol = job['job_type_symbol'] if job['job_type_symbol'] else ''
+                    self.tbplnames.update({testtype: {'group': job_group_symbol,
+                                                      'code': job_type_symbol}})
 
+    # return all jobtypes avaliable on buildbot for now
+    def jobtype_query(self):
+        jobtypes = []
+        for job in self.joblist:
+            testtype = job['ref_data_name'].split(' ')[-1]
+            if testtype in ['mochitest-jetpack', 'dep', 'nightly', 'non-unified',
```

Comment:
we don't have a good simple way to determine if a job is a build or a test
job.  Ideally we can fix that at the runnableAPI level in the future, but
for now we have to look at the list of jobs and have lists to exclude stuff
here and there.

On Tue, Jun 28, 2016 at 6:56 PM, Tiramisu 1993 <notifications@github.com>
wrote:

> In src/treecodes.py
> <https://github.com/mozilla/ouija/pull/93#discussion_r68786209>:
>
> >
> > +    # return all jobtypes avaliable on buildbot for now
> > +    def jobtype_query(self):
> > +        jobtypes = []
> > +        for job in self.joblist:
> > +            testtype = job['ref_data_name'].split(' ')[-1]
> > +            if testtype in ['mochitest-jetpack', 'dep', 'nightly', 'non-unified',
>
> hey Joel (@jmaher <https://github.com/jmaher>), do we have some documents
> about tests(jobtypes)? I don't know how to distinguish which jobtype is for
> buildong and which one is for testing.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/mozilla/ouija/pull/93/files/d30fc202d6c856a3739f2fcf088846e54561856f#r68786209>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAD0fPvR6EJ7ZYO8WHvA0jwl7F2gcZ5oks5qQUQxgaJpZM4I-Umx>
> .
>